### DISCUSSION TITLE
OF Uniformity

### DISCUSSION THREAD

**Discussion by dimitre** [openframeworks/openFrameworks#7039] [General]
Uniformity between different functions, naming, parameter order helps the ease of use.
I'll list some suggestions related to this subject here:

- [x] https://github.com/openframeworks/openFrameworks/pull/6906
- [x] https://github.com/openframeworks/openFrameworks/pull/6944
- [ ] https://github.com/openframeworks/openFrameworks/pull/6939
- [ ] https://github.com/openframeworks/openFrameworks/pull/6940
- [ ] TARGET_OF_IOS, TARGET_OF_WATCHOS, TARGET_ANDROID, TARGET_LINUX, TARGET_OSX
- [ ]  ofStringReplace behaves differently from all other functions (result in a pointer instead of return)
- [ ] https://github.com/openframeworks/openFrameworks/issues/6858
- [x] https://github.com/openframeworks/openFrameworks/issues/6527

**Comment by roymacdonald** [comment]
I think that this should be added to this list as well https://github.com/openframeworks/openFrameworks/issues/6527

**Comment by 2bbb** [comment]
About env detection macro,
`TARGET_OF_IOS`, `TARGET_OF_WATCHOS`, `TARGET_ANDROID`, `TARGET_LINUX`, `TARGET_OSX` and etc, etc...

maybe there is reason, but sometime I confuse about "why not all `TARGET_OF_XXX` or `TARGET_XXX`?"

**Comment by dimitre** [comment]
Other thing I think it is strange is ofRectangle parameters is x, y, width, height (position and dimensions)
but if I use two glm::vec2 to initialize it I'm not talking anymore about position and dimension but of point 1 and point 2 (bounds)

**Comment by dimitre** [comment]
GLFWWindow calls the window dimensions shape or size depending on getting or setting:
setWindowShape / getWindowSize
it also uses different types (glm::vec2 vs int, int)
